packages/navigation/src/NavigationMenuDropdown.tsx

```diff
@@ -14,31 +16,56 @@ type NavigationMenuDropdownProps<T extends object> = CollectionBase<T> & {
 
 const NavigationMenuDropdown = <T extends object>(props: NavigationMenuDropdownProps<T>) => {
   const state = useListState(props);
+  const ref = useRef<HTMLDivElement>(null);
+  // Track the width of the dropdown button so that we can set the menu content
+  // to match: we'll apply this value as min-height to each menu item (so that
+  // the menu item will be at least as wide as the dropdown button, but can
+  // extend beyond it if necessary).
+  const [dropdownButtonWidth, setDropdownButtonWidth] = useState<null | number>(null);
+
+  useEffect(() => {
+    const button = ref.current?.querySelector('button');
+
+    if (!button) {
+      return;
+    }
+
+    setDropdownButtonWidth(button.getBoundingClientRect().width);
+  }, [setDropdownButtonWidth]);
 
   return (
```

Comment:
"Hide Whitespace" helps a bit here:

* Wrap the rendered `Dropdown` in a new `div` with a ref, and add a `placement` prop
* Wrap the `onClick` prop, so that in addition to the event param we can add an additional `{ collapsed: true }` state param